sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/PruningSuite.scala

```diff
@@ -141,6 +141,21 @@ class PruningSuite extends HiveComparisonTest with BeforeAndAfter {
       Seq("2008-04-08", "11"),
       Seq("2008-04-09", "11")))
 
+  createPruningTest("Partition pruning - with filter containing non-deterministic condition",
+    "SELECT value, hr FROM srcpart1 WHERE ds = '2008-04-08' AND hr < 12 AND rand() < 1",
+    Seq("value", "hr"),
+    Seq("value", "hr"),
+    Seq(
+      Seq("2008-04-08", "11")))
+
+  createPruningTest(
+    "Partition pruning - with filter containing non-deterministic condition in sub And-expr",
+    "SELECT value, hr FROM srcpart1 WHERE (ds = '2008-04-08' AND rand() < 1) AND (hr < 12)",
```

Comment:
not sure we need this. As a UT, we know that it doesn't matter how predicates are ANDed together.

Reply:
ok, I will drop.